modules/library/coverage/src/main/java/org/geotools/coverage/processing/operation/Mosaic.java

```diff
@@ -853,6 +869,20 @@ protected RenderedImage createRenderedImage(
         double[][] thresholds = getParameter(parameters, THRESHOLD_PARAM);
         Range[] noData = getParameter(parameters, NODATA_RANGE_PARAM);
         double[] backgrounds = getParameter(parameters, BACKGROUND_PARAM);
+
+        // the background value defaults to <code>new double[] {0}</code>, which makes
+        // the ImageWorker use it and not the the ROIs in output, as they are supposed to
```

Comment:
"the the ROIs in output"

I think they are in input, actually?

"the ROIs in input"

Reply:
Ah, was meant to say "not set the ROIs in output". Fixing.